### ISSUE TITLE
examples, docs and other things

### ISSUE THREAD

**Issue by CaledoniaProject** [gokrazy/rsync#31]
Do you have examples on how to use this package as a rsync client? And it does not seem to be published on pkg.go.dev.

**Comment by stapelberg** [comment]
> Do you have examples on how to use this package as a rsync client? 

The `gokr-rsync` program is a receiver (downloads files). Start at https://github.com/gokrazy/rsync/blob/main/cmd/gokr-rsync/rsync.go and navigate through the code.

Or did you mean something else? In rsync, sender and receiver are the preferred terminology, as client and server can both be either sender or receiver. So, describe in as much detail as possible what you’re looking for.

> And it does not seem to be published on pkg.go.dev.

But it is published on pkg.go.dev? https://pkg.go.dev/github.com/gokrazy/rsync

**Comment by CaledoniaProject** [comment]
Thanks,  I just want to integrate this library into my code (as a client). However, I checked the document, but there isn't much thing exposed right now.

**Comment by stapelberg** [comment]
Yeah, we’d need a refactoring of the client bits into a library. Pull requests welcome :)
